### PR TITLE
Utilize internal typing for number

### PR DESCRIPTION
**Pull request by ros3bud**
- added tests for mortgage.loan.amount to accept range
- updated mortgage.loan.amount type to range for each form
- removed methods parse.integer, parse.float 
- removed occurrences of parseInt and parseFloat 
- made text edits for consistency throughout tests

### CODE REVIEW

**Comment by ros3bud** [comment]
@cgrayson, LGTY?

File: spec/parse_spec.coffee [hunk 1]
```diff
@@ -166,22 +166,22 @@ describe 'Date', ->
 describe 'Integer', ->
 
   it 'should parse number as Integer', ->
-    assert.deepEqual request('mortgage.new_property_value': '290000.00'), 290000
-    assert.deepEqual request('mortgage.new_property_value': '320000'), 320000
+    assert.equal request('mortgage.new_property_value': '290000.00'), '290000.00'
```
Comment (cgrayson) [inline spec/parse_spec.coffee]: For these changes, just one or the other would have been enough: either change from `deepEqual` to `equal`, or change the expected value from numeric to string. Though technically it is indeed a string coming back from `request()`, it looks weird to me that the test "it 'should parse number as Integer'" is comparing to a string.

File: spec/unified_long_homeequity.coffee [hunk 2]
```diff
@@ -91,22 +91,25 @@ describe 'LongForm HomeEquity', ->
...
-      it 'should describe loan cash out', ->
+      it 'should parse loan cash out', ->
         assert.equal unified.request(req(xml())).mortgage.loan.cash_out, null
 
-      it 'should describe loan purpose', ->
+      it 'should parse loan purpose', ->
         assert.equal unified.request(req(xml())).mortgage.loan.purpose, 'equity'
 
-      xit 'should describe loan ltv', ->
+      xit 'should parse loan ltv', ->
```
Comment (cgrayson) [inline spec/unified_long_homeequity.coffee]: Appreciate the consistency :smiley:, but maybe take this opportunity to either delete or unpause `xit`ed tests.

File: src/parse.coffee [hunk 3]
```diff
@@ -95,23 +95,14 @@ parseDate = (value, format='MM/DD/YYYY') ->
...
-  else
-    null
+# parseFloat = (value) ->
+  # return null unless value?.trim()
+  # value = value.replace(/[^0-9.]/g, '')
+  # return null if isNaN(value)
+  # if value.trim().length > 0
+    # parseFloat(value)
+  # else
+    # null
```
Comment (cgrayson) [inline src/parse.coffee]: Is there a reason to keep this commented-out code?

**Comment by cgrayson** [comment]
Let me know if you have further changes you want me to glance over, but other than the couple comments above, LGTM.

**Comment by ros3bud** [comment]
Thanks so much for the feedback - will let you know if anything else comes up, but this all makes good sense to me.

File: spec/parse_spec.coffee (hunk 1 above)
Comment (ros3bud) [inline spec/parse_spec.coffee]: Thinking I will go with the deepEquals and update the it statement to: "  it 'should parse numbers appropriately for internal number type', ->"

File: spec/unified_long_homeequity.coffee (hunk 2 above)
Comment (ros3bud) [inline spec/unified_long_homeequity.coffee]: Lol, yeah, I suppose that makes sense...

**Comment by ros3bud** [comment]
updated type back to number, removed range tests

@alexkwolfe, LGTY?

File: spec/longform_spec.coffee [hunk 4]
```diff
@@ -120,10 +120,10 @@ describe 'Longform', ->
     describe 'Loan', ->
 
       it 'should parse amount', ->
-        assert.equal longform.request(req(xml())).mortgage.loan.amount, 800000
+        assert.equal longform.request(req(xml())).mortgage.loan.amount, 800000.88
 
       it 'should parse down payment', ->
-        assert.equal longform.request(req(xml())).mortgage.loan.down_payment, 200000
+        assert.equal longform.request(req(xml())).mortgage.loan.down_payment, 200000.20
```
Comment (alexkwolfe) [inline spec/longform_spec.coffee]: Should this be testing for a string?

File: spec/unified_long_personal.coffee [hunk 5]
```diff
@@ -86,24 +86,21 @@ describe 'LongForm Personal', ->
         assert.equal unified.request(req(xml())).mortgage.property.purchase_year, null
 
     describe 'Loan', ->
-      it 'should describe loan term', ->
+      it 'should parse loan term', ->
         assert.equal unified.request(req(xml())).mortgage.loan.term, null
 
-      it 'should describe loan amount', ->
+      it 'should parse loan amount', ->
         assert.equal unified.request(req(xml())).mortgage.loan.amount, 3250000
```
Comment (alexkwolfe) [inline spec/unified_long_personal.coffee]: Should this be testing for a string?

**Comment by alexkwolfe** [comment]
Other that the aforementioned issues with string vs integer (and there are a few others in this integration I didn't bother commenting on), LGTM

**Comment by ros3bud** [comment]
Thanks for catching that - I updated those and the others I saw.

**Pull request merged**